app/Services/Patient/PatientService.cs

```diff
@@ -0,0 +1,120 @@
+            }
+        }
+
+        private PatientResponse EditPatientInternal(PatientModel patient, EditPatientRequest request)
+        {
+            this.ValidatePatient(request.Email, request.Pesel);
+
+            if (request.Address != null)
+            {
+                var address = request.Address;
+                if (address.Street != null)
+                    patient.Address.Street = address.Street;
+
+                if (address.City != null)
+                    patient.Address.City = address.City;
+
+                if (address.ZipCode != null)
+                    patient.Address.ZipCode = address.ZipCode;
+
+                if (address.HouseNumber != null)
+                    patient.Address.HouseNumber = address.HouseNumber;
+
+                if (address.LocalNumber != null)
+                    patient.Address.LocalNumber = address.LocalNumber;
+            }
+
+            if (request.FirstName != null)
+                patient.FirstName = request.FirstName;
+
+            if (request.LastName != null)
+                patient.LastName = request.LastName;
+
+            if (request.Pesel != null)
+                patient.Pesel = request.Pesel;
+
+            if (request.Email != null)
+                patient.Email = request.Email;
+
+            this.dataContext.Update(patient);
+            this.dataContext.SaveChanges();
+
+            return new PatientResponse(patient);
+        }
+
+        public PatientService(DataContext dataContext, SecurePasswordHasher securePasswordHasher)
+        {
+            this.dataContext = dataContext;
+            this.securePasswordHasher = securePasswordHasher;
+        }
+
+        public async Task<PatientResponse> Register(PatientRegistrationRequest request)
+        {
+            this.ValidatePatient(request.Email, request.Pesel);
+
+            // Add patient to database
+            AddressModel address = new AddressModel()
+            {
+                City = request.Address.City,
+                ZipCode = request.Address.ZipCode,
+                Street = request.Address.Street,
+                HouseNumber = request.Address.HouseNumber,
+                LocalNumber = request.Address.LocalNumber
+            };
+
+            PatientModel patient = new PatientModel()
+            {
+                FirstName = request.FirstName,
+                LastName = request.LastName,
+                Email = request.Email,
+                Password = this.securePasswordHasher.Hash(request.Password),
+                Address = address,
+                Pesel = request.Pesel
+            };
+
+            this.dataContext.Patients.Add(patient);
+            this.dataContext.SaveChanges();
+
+            return new PatientResponse(patient);
+        }
+
+        public async Task<PatientResponse> EditPatient(PatientModel patient, EditPatientRequest request) 
+            => this.EditPatientInternal(patient, request);
```

Comment:
Tu miało jakiś konkretny cel zrealizowanie tego za pomocą prywatnej metody zamiast zostawienia tego w tej metodzie i w tej drugiej z patientId wywołania tej metody?

Reply:
tak

z jednej strony mamy request admina, który ma tylko id pacjenta, więc musimy zakwerendować o obiekt
z drugiej strony mamy request pacjenta, gdzie obiekt już jest odczytany z bazy podczas walidacji tokena

żeby nie kwerendować o obiekt dwa razy w drugim przypadku zrobiłem wrappery - jeden ma już pacjenta, drugi kwerenduje o niego do db, potem odbywa się wspólna część walidacyjna i update pacjenta

Reply:
Nie o to mi chodziło. Chodziło mi o obecność tej prywatnej metody EditPatientInternal, równie dobrze to wszystko co w niej jest mogłoby być w metodzie EditPatient(PatienModel patient ...) i metoda EditPatient(int patientId ... nie wywowyłała metody EditPatientInternal tylko metodę EditPatient(PatientModel patient...)

Reply:
ok. można tak zrobić, dla mnie bardziej logiczny byłby podział dwa wrappery i jeden handler, ale można to tak zmienić dla przejrzystości

Reply:
zrobione